crates/proto/src/rr/rdata/cert.rs

```diff
@@ -0,0 +1,708 @@
+        key_tag: u16,
+        algorithm: Algorithm,
+        cert_data: Vec<u8>,
+    ) -> Self {
+        Self {
+            cert_type,
+            key_tag,
+            algorithm,
+            cert_data,
+        }
+    }
+
+    /// Returns the CERT type
+    pub fn cert_type(&self) -> CertType {
+        self.cert_type
+    }
+
+    /// Returns the CERT key tag
+    pub fn key_tag(&self) -> u16 {
+        self.key_tag
+    }
+
+    /// Returns the CERT algorithm
+    pub fn algorithm(&self) -> Algorithm {
+        self.algorithm
+    }
+
+    /// Returns the CERT record data
+    pub fn cert_data(&self) -> Vec<u8> {
+        self.cert_data.clone()
+    }
+
+    /// Returns the CERT (Base64)
+    pub fn cert_base64(&self) -> String {
+        data_encoding::BASE64.encode(&self.cert_data).clone()
+    }
+}
+
+impl TryFrom<&[u8]> for CERT {
+    type Error = ProtoError;
+
+    fn try_from(cert_record: &[u8]) -> Result<Self, Self::Error> {
+        if cert_record.len() <= 5 {
+            return Err(ProtoError::from("invalid cert_record length".to_string()));
+        }
+
+        let cert_type = ((cert_record[0] as u16) << 8) | (cert_record[1] as u16);
+        let key_tag = ((cert_record[2] as u16) << 8) | (cert_record[3] as u16);
+        let algorithm = cert_record[4];
+        let cert_data = cert_record[5..].to_vec();
+
+        Ok(Self {
+            cert_type: cert_type.into(),
+            key_tag,
+            algorithm: algorithm.into(),
+            cert_data,
+        })
+    }
+}
+
+impl BinEncodable for CERT {
+    fn emit(&self, encoder: &mut BinEncoder<'_>) -> ProtoResult<()> {
+        encoder.emit_u16(self.cert_type.into())?;
+        encoder.emit_u16(self.key_tag)?;
+        encoder.emit_u8(self.algorithm.into())?;
+        encoder.emit_vec(&self.cert_data)?;
+
+        Ok(())
+    }
+}
+
+impl<'r> RecordDataDecodable<'r> for CERT {
+    fn read_data(decoder: &mut BinDecoder<'r>, length: Restrict<u16>) -> ProtoResult<Self> {
+        let rdata_length = length.map(|u| u as usize).unverified();
+        if rdata_length > 0 {
+            let cert_data = decoder.read_vec(rdata_length)?.unverified(/*any byte array is good*/);
+            Ok(Self::try_from(&cert_data[..])?)
+        } else {
+            Ok(Self::new(
+                CertType::Reserved,
+                0,
+                Algorithm::Reserved(0),
+                vec![],
+            ))
+        }
+    }
+}
+
+impl RecordData for CERT {
+    fn try_from_rdata(data: RData) -> Result<Self, RData> {
+        match data {
+            RData::CERT(data) => Ok(data),
+            _ => Err(data),
+        }
+    }
+
+    fn try_borrow(data: &RData) -> Option<&Self> {
+        match data {
+            RData::CERT(data) => Some(data),
+            _ => None,
+        }
+    }
+
+    fn record_type(&self) -> RecordType {
+        RecordType::CERT
+    }
+
+    fn into_rdata(self) -> RData {
+        RData::CERT(self)
+    }
+}
+
+impl fmt::Display for CERT {
```

Comment:
I like to drop the RFCs presentation format section into the docs here, it makes it easy to review the format:

```text
[2.2](https://datatracker.ietf.org/doc/html/rfc4398#section-2.2).  Text Representation of CERT RRs

   The RDATA portion of a CERT RR has the type field as an unsigned
   decimal integer or as a mnemonic symbol as listed in [Section 2.1](https://datatracker.ietf.org/doc/html/rfc4398#section-2.1),
   above.

   The key tag field is represented as an unsigned decimal integer.

   The algorithm field is represented as an unsigned decimal integer or
   a mnemonic symbol as listed in [[12](https://datatracker.ietf.org/doc/html/rfc4398#ref-12)].

   The certificate/CRL portion is represented in base 64 [[16](https://datatracker.ietf.org/doc/html/rfc4398#ref-16)] and may be
   divided into any number of white-space-separated substrings, down to
   single base-64 digits, which are concatenated to obtain the full
   signature.  These substrings can span lines using the standard
   parenthesis.

   Note that the certificate/CRL portion may have internal sub-fields,
   but these do not appear in the master file representation.  For
   example, with type 254, there will be an OID size, an OID, and then
   the certificate/CRL proper.  However, only a single logical base-64
   string will appear in the text representation.
```